pkg/controllers/provisioningv2/fleetcluster/fleetcluster.go

```diff
@@ -125,6 +126,7 @@ func (h *handler) createCluster(cluster *v1.Cluster, status v1.ClusterStatus) ([
 			KubeConfigSecret: clientSecret,
 			AgentEnvVars:     mgmtCluster.Spec.AgentEnvVars,
 			AgentNamespace:   agentNamespace,
+			PrivateRepoURL:   image.GetPrivateRepoURLFromCluster(cluster),
```

Comment:
I believe we will run into an issue with this, as this only will detect a private registry if it is either set as a system default registry or via the machineGlobalData/machineConfigData. It will not detect a cluster-level registry

@HarrisonWAffel just went through the exercise of figuring out which method can be used to find the "correct" private repo -- perhaps we can use the logic from that over here.

Reply:
I think that this is actually OK for RKE2/K3s clusters. When creating a RKE2/K3s cluster via the UI and supplying a cluster-level registry, the URL will always be placed in the `machineSelectorConfig`. This function was also used in the other PR for the cluster agent manifest, and has used to [determine the desired version of the agent ](https://github.com/HarrisonWAffel/rancher/blob/c0ba0f0c94a1e43885a6821a5403c5d600ef1246/pkg/controllers/provisioningv2/cluster/controller.go#L277) for a little while now.

Reply:
@HarrisonWAffel ok, thank you for the clarification.

Do you know what the purpose of the registry field on the spec is then and when that would be set? I'm worried we will have users setting that instead of setting the machineSelectorConfig -- we can't always assume a user will mimic what the UI does.

Reply:
@Oats87 Right now the `registries` section is used to hold registry auth configs, and has a similar format to `registries.yaml`, here's what it looks like when created by rancher without tls enabled
```
    registries:
      configs:
        [PRIVATE_REGISTRY_URL]:
          authConfigSecretName: registryconfig-auth-5fqst
          caBundle: null
          insecureSkipVerify: false
          tlsSecretName: null
```
So, while the `registries` section can contain authentication configs for multiple registries, the `machineSelectorConfig` will denote the system-default registry where system-images should be pulled from.

Reply:
Makes sense -- thanks for the clarification!

In this case though, do we need to support authenticating against the private registry being used @manno @mattfarina ?

I'm not sure if Fleet supports setting an image pull secret or something along that path.

Reply:
@Oats87 Fleet doesn't support auth at the moment. I've updated the existing issue for us to add it. Hopefully for 2.7.1.

Do we need auth for 2.7.0 is more of a question for mgmt.

Reply:
What happens if you have a cluster registry with auth? Does it fail?